[docs] List all pickers toolbar pages in api docs side menu

LukasTy opened the pull request:
Noticed that we are changing contents of these pages quite often, but they are not listed anywhere.
We did list `DateTimePickerTabs` component, I do not see a reason to not list these components.

flaviendelangle reviewed (approved):
Nice catch
They should probably also be listed in `api/index.md`

LukasTy commented:
> They should probably also be listed in `api/index.md`

Yup. I've added those in `Toolbars` group.
I've also added the `Index` Api reference submenu item as the first child—same as in Data Grid api docs. 😉

flaviendelangle commented on docs/pages/x/api/date-pickers/index.md:
```diff
@@ -55,6 +55,13 @@
 
+### Toolbars
+
+- [DatePickerToolbar](/x/api/date-pickers/date-picker-toolbar/)
```
Alternatively we could put it next to their picker (like for the tabs)

If we keep them here, could we keep the regular order "Date => Time => Date Time => Date Range" that we tend to use most of the time.
